hyperspy/gui/egerton_quantification.py

```diff
@@ -369,10 +369,12 @@ def _interpolator_kind_changed(self, old, new):
         self.span_selector_changed()
 
     def _ss_left_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
+            self.span_selector_changed()
 
     def _ss_right_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
```

Comment:
Ah, so this code basically does two things then:
 - If the values were already NaN, and left is being set, it does not trigger the event since right is still NaN. When right is then set, it triggers the event. This point presupposes that the left value is always set before the right value.
 - If being reset, it should prevent the event from triggering as well. However, as far as I can tell, this should presuppose that the right is being set first, otherwise the event will trigger when the left value is being set to NaN, and currently the reset function sets the left value first.

A more robust way of checking this would perhaps be to check in both left and right `value_changed` functions whether _either_ of the two values are NaN?